packages/react-query/src/queryOptions.ts

```diff
@@ -32,26 +32,26 @@ export function queryOptions<
   TError = DefaultError,
   TData = TQueryFnData,
   TQueryKey extends QueryKey = QueryKey,
-  TOptions extends UndefinedInitialDataOptions<
-    TQueryFnData,
-    TError,
-    TData,
-    TQueryKey
-  > = UndefinedInitialDataOptions<TQueryFnData, TError, TData, TQueryKey>,
->(options: ValidateQueryOptions<TOptions>): TOptions
```

Comment:
removing this generic here made this test fail:

https://github.com/TanStack/query/blob/cc9bab714a5b520f37ac1460d9d564c5f0fda87f/packages/react-query/src/__tests__/useQuery.types.test.tsx#L48-L59

However, it made other things a lot easier 😅 . For example, I couldn't get ` DataTag<TQueryKey, TData>` to infer correctly - I had to re-extract the types from `TOptions['queryKey']` etc. Basically the changes from this commit:

- https://github.com/TanStack/query/pull/6119/commits/37f2a8bb4633ba2193e6eb14e262e0eaad584810

So, I'm considering keeping this as a known limitation, because the workaround for that situation would be to change `options` to take select as an arg:

```
const options = <TData = number,>(select?: (data: number) => TData) =>
  queryOptions({
    queryKey: ['key'],
    queryFn: () => Promise.resolve(1),
    select,
  })
useQuery(options((data) => data > 1))
```

which is not that bad. what do you think @Andarist ?